timbermill-java/timbermill-local/src/test/java/com/datorama/oss/timbermill/common/disk/SQLJetDiskHandlerTest.java

```diff
@@ -156,32 +164,119 @@ public void testMultiThreadSafety() throws MaximumInsertTriesException, Interrup
 			executorService.execute(fetchAndPersistTask);
 		}
 
-        Thread.sleep(2000);
+		Thread.sleep(2000);
 
 		// stop threads and wait
-        keepExecuting.set(false);
-        executorService.shutdown();
-        executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
+		keepExecuting.set(false);
+		executorService.shutdown();
+		executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
 
 		assertFalse(isHealthCheckFailed.get());
 	}
 
+	@Test
+	public void validateEventsDeserialization() throws Exception {
+		// Checking if can deserialize previous version of Event
+		boolean deserializationSuccess = true;
+
+		Path path = Paths.get(SQLJetDiskHandlerTest.class.getResource(OLD_VERSION_EVENT).toURI());
+		byte[] oldVersionEventBytes = Files.readAllBytes(path);
+		try {
+			sqlJetDiskHandler.deserializeEvents(oldVersionEventBytes);
+		} catch (SerializationException e) {
+			deserializationSuccess = false;
+		}
+
+		String message = "Failed to deserialize previous version of Event, which may break " +
+				"the connection with the db.\nYou may changed a field's type.";
+		assertTrue(message, deserializationSuccess);
+	}
+
+	@Test
+	public void validateEventsFieldNames() {
```

Comment:
after thinking about it again, maintaining this would be tedious.
since we are deleting badly serialized data now (@edengal 's change) no real need to to have this test.

Reply:
yes, it makes sense
so removed this test @DvirKatz